code/game/machinery/doors/firedoor.dm

```diff
@@ -501,7 +512,70 @@ var/global/list/alert_overlays_global = list()
 			return !density
 	return 1
 
+/obj/machinery/door/firedoor/border_only/is_fulltile()
+	return 0
+
 
 /obj/machinery/door/firedoor/multi_tile
 	icon = 'icons/obj/doors/DoorHazard2x1.dmi'
 	width = 2
+
+
+/obj/item/firedoor_frame
+	name = "firedoor frame"
+	icon = 'icons/obj/doors/DoorHazard.dmi'
+	icon_state = "firedoor_frame"
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/electronics.dmi', "right_hand" = 'icons/mob/in-hand/right/electronics.dmi')
+	desc = "The frame for a firedoor, strangely easy to set up considering its application."
+
+/obj/item/firedoor_frame/attack_self(mob/user)
+	construct_firedoor(user)
+
+/obj/item/firedoor_frame/proc/construct_firedoor(mob/user)
+	if(!user || !src)
+		return 0
+	if(!istype(user.loc,/turf))
+		return 0
+	if(!user.IsAdvancedToolUser())
+		to_chat(user, "<span class='warning'>You don't have the dexterity to do this!</span>")
+		return 0
+	switch(alert("firedoor construction", "Would you like to construct a full tile firedoor or one direction?", "One Direction", "Full Firedoor", "Cancel", null))
+		if("One Direction")
+			if(!src)
+				return 1
+			if(src.loc != user)
+				return 1
+			var/list/directions = new/list(cardinal)
```

Comment:
This is literally window placement code.

Two birds with one stone, we make a good way of doing firedoors, and we make better window code